<a href="https://www.kaggle.com/code/rajatraj0502/layoffs-dataset-2023?scriptVersionId=142131492" target="_blank"><img align="left" alt="Kaggle" title="Open in Kaggle" src="https://kaggle.com/static/images/open-in-kaggle.svg"></a>

In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/layoffs-data-2022/layoffs_data.csv


In [2]:
# Loading the dataset
df = pd.read_csv('/kaggle/input/layoffs-data-2022/layoffs_data.csv')

# Displaying the first few rows
df.head()


Company  Location_HQ        Industry  Percentage        Date  \
0        IRL  SF Bay Area        Consumer        1.00  2023-06-23   
1     Retool  SF Bay Area           Other        0.09  2023-06-22   
2    Anaplan  SF Bay Area           Other         NaN  2023-06-21   
3       Uber  SF Bay Area  Transportation         NaN  2023-06-21   
4  Tackle.io        Boise  Infrastructure         NaN  2023-06-21   

                                              Source  Funds_Raised     Stage  \
0  https://www.theinformation.com/articles/social...         197.0  Series C   
1                                      Internal memo         141.0  Series C   
2  https://nypost.com/2023/06/23/software-giant-a...         300.0  Acquired   
3  https://www.nasdaq.com/articles/uber-to-lay-of...       25200.0  Post-IPO   
4      https://tackle.io/blog/tackle-company-update/         148.0  Series C   

            Date_Added        Country  Laid_Off_Count  \
0  2023-06-25 23:04:16  United States             NaN   
1  2023-06-22 14:23:40  United States             NaN   
2  2023-06-24 03:48:00  United States           300.0   
3  2023-06-22 00:19:03  United States           200.0   
4  2023-06-23 15:38:59  United States            75.0   

  List_of_Employees_Laid_Off  
0                    Unknown  
1                    Unknown  
2                    Unknown  
3                    Unknown  
4                    Unknown

**1. Bar chart: Top Industries with the Most Layoffs**

In [3]:
import plotly.express as px

# Grouping by industry and summing the layoffs
industry_layoffs = df.groupby('Industry')['Laid_Off_Count'].sum().sort_values(ascending=False).head(10)

# Bar chart for top industries with the most layoffs
fig1 = px.bar(industry_layoffs, 
              title='Top Industries with the Most Layoffs', 
              labels={'Industry': 'Industry', 'Laid_Off_Count': 'Number of Layoffs'},
              color_discrete_sequence=px.colors.sequential.RdBu)

fig1.show()


**2. Time Series Plot: Trend of Layoffs Over Time**

In [4]:
# Convert the 'Date' column to a datetime format
df['Date'] = pd.to_datetime(df['Date'])

# Group by date and sum the layoffs
date_layoffs = df.groupby('Date')['Laid_Off_Count'].sum()

# Time series plot for layoffs over time
fig2 = px.line(date_layoffs.reset_index(),
               x='Date',
               y='Laid_Off_Count',
               title='Trend of Layoffs Over Time',
               labels={'Date': 'Date', 'Laid_Off_Count': 'Number of Layoffs'},
               color_discrete_sequence=px.colors.sequential.RdBu)

fig2.show()


**3. Pie Chart: Distribution of Company Stages Among the Layoffs**

In [5]:
# Group by stage and sum the layoffs
stage_layoffs = df.groupby('Stage')['Laid_Off_Count'].sum()

# Pie chart for distribution of company stages among the layoffs
fig3 = px.pie(stage_layoffs.reset_index(),
              names='Stage',
              values='Laid_Off_Count',
              title='Distribution of Company Stages Among the Layoffs',
              color_discrete_sequence=px.colors.sequential.RdBu)

fig3.show()


**4. Bar Chart: Companies with the Highest Number of Layoffs**

In [6]:
# Group by company and sum the layoffs
company_layoffs = df.groupby('Company')['Laid_Off_Count'].sum().sort_values(ascending=False).head(10)

# Bar chart for companies with the highest number of layoffs
fig4 = px.bar(company_layoffs.reset_index(),
              x='Company',
              y='Laid_Off_Count',
              title='Companies with the Highest Number of Layoffs',
              labels={'Company': 'Company', 'Laid_Off_Count': 'Number of Layoffs'},
              color='Laid_Off_Count',
              color_continuous_scale=px.colors.sequential.RdBu)

fig4.show()


**5. Map Visualization: Highlighting Locations with the Highest Layoffs**

In [7]:
# Group by country and sum the layoffs
country_layoffs = df.groupby('Country')['Laid_Off_Count'].sum()

# Map visualization for countries with layoffs
fig5 = px.choropleth(country_layoffs.reset_index(), 
                     locations='Country',
                     locationmode='country names',
                     color='Laid_Off_Count',
                     hover_name='Country',
                     color_continuous_scale=px.colors.sequential.RdBu,
                     title='Number of Layoffs by Country')

fig5.show()


**6. Histogram: Distribution of Funds Raised by Companies**

In [8]:
# Histogram for distribution of funds raised by companies
fig6 = px.histogram(df,
                    x='Funds_Raised',
                    title='Distribution of Funds Raised by Companies',
                    color_discrete_sequence=px.colors.sequential.RdBu,
                    labels={'Funds_Raised': 'Amount of Funds Raised'},
                    nbins=30)

fig6.show()


**7. Bar Chart: Top Locations with Most Layoffs**

In [9]:
# Group by location and sum the layoffs
location_layoffs = df.groupby('Location_HQ')['Laid_Off_Count'].sum().sort_values(ascending=False).head(10)

# Bar chart for top locations with the most layoffs
fig7 = px.bar(location_layoffs.reset_index(), 
              x='Location_HQ', 
              y='Laid_Off_Count', 
              title='Top Locations with Most Layoffs', 
              labels={'Location_HQ': 'Location', 'Laid_Off_Count': 'Number of Layoffs'},
              color='Laid_Off_Count',
              color_continuous_scale=px.colors.sequential.RdBu)

fig7.show()


**8. Scatter Plot: Funds Raised vs. Number of Layoffs**

In [10]:
# Scatter plot for funds raised vs number of layoffs
fig8 = px.scatter(df, 
                  x='Funds_Raised', 
                  y='Laid_Off_Count', 
                  title='Funds Raised vs. Number of Layoffs', 
                  labels={'Funds_Raised': 'Funds Raised ($)', 'Laid_Off_Count': 'Number of Layoffs'},
                  color='Laid_Off_Count',
                  color_continuous_scale=px.colors.sequential.RdBu,
                  hover_name='Company')

fig8.show()


**9. Bar Chart: Layoffs Distribution by Stage**

In [11]:
# Group by stage and sum the layoffs
stage_layoffs_count = df.groupby('Stage')['Laid_Off_Count'].sum().sort_values(ascending=False)

# Bar chart for layoffs distribution by stage
fig9 = px.bar(stage_layoffs_count.reset_index(), 
              x='Stage', 
              y='Laid_Off_Count', 
              title='Layoffs Distribution by Stage', 
              labels={'Stage': 'Company Stage', 'Laid_Off_Count': 'Number of Layoffs'},
              color='Laid_Off_Count',
              color_continuous_scale=px.colors.sequential.RdBu)

fig9.show()


**10. Box Plot: Distribution of Layoffs Across Industries**

In [12]:
# Box plot for distribution of layoffs across industries
fig10 = px.box(df, 
               x='Industry', 
               y='Laid_Off_Count',
               title='Distribution of Layoffs Across Industries',
               labels={'Industry': 'Industry', 'Laid_Off_Count': 'Number of Layoffs'},
               color_discrete_sequence=px.colors.sequential.RdBu)

fig10.show()


**11. Country-wise Bar Chart: Total Layoffs**

In [13]:
# Group by country and sum the layoffs
country_total_layoffs = df.groupby('Country')['Laid_Off_Count'].sum().sort_values(ascending=False)

# Bar chart for total layoffs by country
fig11 = px.bar(country_total_layoffs.reset_index(), 
               x='Country', 
               y='Laid_Off_Count', 
               title='Total Layoffs by Country', 
               labels={'Country': 'Country', 'Laid_Off_Count': 'Total Layoffs'},
               color='Laid_Off_Count',
               color_continuous_scale=px.colors.sequential.RdBu)

fig11.show()


**12. Country-wise Box Plot: Layoffs Distribution**

In [14]:
# Box plot for distribution of layoffs across countries
fig12 = px.box(df, 
               x='Country', 
               y='Laid_Off_Count',
               title='Distribution of Layoffs Across Countries',
               labels={'Country': 'Country', 'Laid_Off_Count': 'Number of Layoffs'},
               color_discrete_sequence=px.colors.sequential.RdBu)

fig12.show()


**13. Country-wise Bar Chart: Average Funds Raised**

In [15]:
# Group by country and calculate the average funds raised
country_avg_funds = df.groupby('Country')['Funds_Raised'].mean().sort_values(ascending=False)

# Bar chart for average funds raised by country
fig14 = px.bar(country_avg_funds.reset_index(), 
               x='Country', 
               y='Funds_Raised', 
               title='Average Funds Raised by Country', 
               labels={'Country': 'Country', 'Funds_Raised': 'Average Funds Raised ($)'},
               color='Funds_Raised',
               color_continuous_scale=px.colors.sequential.RdBu)

fig14.show()


In [16]:
# Mapping countries to their respective continents
continent_mapping = {
    'North America': ['United States', 'Canada', 'Mexico'],
    'South America': ['Brazil', 'Argentina', 'Colombia', 'Chile', 'Peru', 'Venezuela', 'Ecuador', 'Bolivia', 'Paraguay', 'Uruguay', 'Guyana', 'Suriname'],
    'Asia': ['India', 'China', 'Japan', 'South Korea', 'Indonesia', 'Saudi Arabia', 'Turkey', 'Iran', 'Israel', 'Thailand', 'Philippines', 'Malaysia', 'Vietnam', 'Singapore'],
    'Europe': ['United Kingdom', 'Germany', 'France', 'Italy', 'Spain', 'Netherlands', 'Belgium', 'Poland', 'Sweden', 'Austria', 'Switzerland', 'Norway', 'Denmark', 'Greece', 'Ireland', 'Portugal', 'Finland', 'Ukraine', 'Czech Republic', 'Romania', 'Hungary', 'Bulgaria', 'Croatia', 'Luxembourg', 'Slovakia', 'Belarus', 'Estonia', 'Latvia', 'Lithuania', 'Slovenia', 'Cyprus', 'Malta', 'Iceland'],
    'Africa': ['South Africa', 'Nigeria', 'Egypt', 'Algeria', 'Morocco', 'Kenya', 'Uganda', 'Angola', 'Zimbabwe', 'Tanzania', 'Ghana', 'Ethiopia', 'Madagascar', 'Cameroon', 'Côte d\'Ivoire', 'Senegal', 'Mali', 'Zambia'],
    'Oceania': ['Australia', 'New Zealand'],
    'Antarctica': ['Antarctica']
}

# Reverse the mapping to get country to continent mapping
country_to_continent = {}
for continent, countries in continent_mapping.items():
    for country in countries:
        country_to_continent[country] = continent

# Add a new column 'Continent' to the dataframe
df['Continent'] = df['Country'].map(country_to_continent)

# Aggregate the layoff data at the continent level
continent_layoffs = df.groupby('Continent')['Laid_Off_Count'].sum()
country_layoffs = df.groupby(['Continent', 'Country'])['Laid_Off_Count'].sum()

continent_layoffs, country_layoffs.head()


(Continent
 Africa             2537.0
 Asia              64484.0
 Europe            61280.0
 North America    325658.0
 Oceania            4111.0
 South America     11781.0
 Name: Laid_Off_Count, dtype: float64,
 Continent  Country
 Africa     Egypt         0.0
            Ghana         0.0
            Kenya       355.0
            Nigeria    1882.0
            Senegal     300.0
 Name: Laid_Off_Count, dtype: float64)

**Enhanced Pie Chart: Distribution of Company Stages Among the Layoffs**

In [17]:
import plotly.graph_objects as go

# Group by stage and sum the layoffs
stage_layoffs = df.groupby('Stage')['Laid_Off_Count'].sum()

# Custom color palette
colors = ['#1f77b4', '#ff7f0e', '#2ca02c', '#d62728', '#9467bd', '#8c564b', '#e377c2', '#7f7f7f', '#bcbd22', '#17becf']

# Enhanced pie chart
fig15 = go.Figure(data=[go.Pie(labels=stage_layoffs.index,
                               values=stage_layoffs.values,
                               textinfo='percent+label',
                               insidetextorientation='radial',
                               marker=dict(colors=colors, line=dict(color='#000000', width=2)))],
                  layout=go.Layout(title='Distribution of Company Stages Among the Layoffs'))

# Display the pie chart
fig15.show()


**Bar Chart: Top Industries with the Most Layoffs**

In [18]:
# Grouping by industry and summing the layoffs
industry_layoffs = df.groupby('Industry')['Laid_Off_Count'].sum().sort_values(ascending=False).head(10)

# Enhanced bar chart for top industries with the most layoffs
fig16 = go.Figure([go.Bar(x=industry_layoffs.index,
                          y=industry_layoffs.values,
                          text=industry_layoffs.values,
                          textposition='auto',
                          marker=dict(color=colors, line=dict(color='#000000', width=1)))],
                  layout=go.Layout(title='Top Industries with the Most Layoffs'))

fig16.show()


In [19]:
# Enhanced scatter plot for layoffs vs. funds raised colored by industry
fig19 = px.scatter(df,
                   x='Funds_Raised',
                   y='Laid_Off_Count',
                   color='Industry',
                   trendline='ols',
                   hover_name='Company',
                   labels={'Funds_Raised': 'Funds Raised ($)', 'Laid_Off_Count': 'Number of Layoffs'},
                   title='Layoffs vs. Funds Raised by Industry')
fig19.update_traces(marker=dict(line=dict(color='#000000', width=1)))
fig19.show()


In [20]:
# Calculate cumulative layoffs over time
df_sorted_date = df.sort_values(by='Date')
df_sorted_date['Cumulative_Layoffs'] = df_sorted_date['Laid_Off_Count'].cumsum()

# Enhanced time series for cumulative layoffs over time
fig21 = px.line(df_sorted_date,
                x='Date',
                y='Cumulative_Layoffs',
                title='Cumulative Layoffs Over Time',
                labels={'Date': 'Date', 'Cumulative_Layoffs': 'Cumulative Number of Layoffs'})
fig21.update_traces(line=dict(color='#1f77b4', width=3))
fig21.show()


In [21]:
# Grouping data by Date and Industry and summing the layoffs
industry_date_layoffs = df.groupby(['Date', 'Industry'])['Laid_Off_Count'].sum().reset_index()

# Enhanced stacked bar chart for layoffs by industry over time
fig22 = px.bar(industry_date_layoffs,
               x='Date',
               y='Laid_Off_Count',
               color='Industry',
               title='Layoffs by Industry Over Time',
               labels={'Date': 'Date', 'Laid_Off_Count': 'Number of Layoffs'},
               color_discrete_sequence=colors)
fig22.show()


In [22]:
# Enhanced violin plot for distribution of layoffs across industries
fig24 = px.violin(df,
                  y='Laid_Off_Count',
                  x='Industry',
                  box=True,
                  points="all",
                  title='Distribution of Layoffs Across Industries',
                  color_discrete_sequence=colors)
fig24.show()



In [23]:
# Loading the dataset
df = pd.read_csv('/kaggle/input/layoffs-data-2022/layoffs_data.csv')

# Filtering the dataset for entries related to India
india_df = df[df['Country'] == 'India']

# Displaying the first few rows of the India-specific data
india_df.head()


Company Location_HQ    Industry  Percentage        Date  \
10    Chigari   Bengaluru       Media         0.2  2023-06-19   
13   Mojocare   Bengaluru  Healthcare         0.8  2023-06-17   
21   FrontRow   Bengaluru   Education         0.9  2023-06-15   
26  Mamaearth    Gurugram      Retail         NaN  2023-06-14   
45     Byju's   Bengaluru   Education         NaN  2023-06-07   

                                               Source  Funds_Raised  \
10  https://inc42.com/buzz/exclusive-weeks-after-c...          88.0   
13  https://entrackr.com/2023/06/exclusive-mojocar...          24.0   
21  https://inc42.com/buzz/lightspeed-backed-edtec...          17.0   
26  https://inc42.com/buzz/exclusive-ipo-bound-mam...         111.0   
45  https://themorningcontext.com/internet/byjus-t...        5500.0   

             Stage           Date_Added Country  Laid_Off_Count  \
10         Unknown  2023-06-20 02:56:18   India            48.0   
13        Series A  2023-06-17 23:41:14   India           170.0   
21        Series A  2023-06-16 04:30:29   India             NaN   
26         Unknown  2023-06-14 18:29:59   India            80.0   
45  Private Equity  2023-06-09 18:57:11   India          1000.0   

   List_of_Employees_Laid_Off  
10                    Unknown  
13                    Unknown  
21                    Unknown  
26                    Unknown  
45                    Unknown

In [24]:
# Overview of the data for India

# Number of unique companies in India that had layoffs
num_companies = india_df['Company'].nunique()

# Total layoffs in India
total_layoffs = india_df['Laid_Off_Count'].sum()

# Average funds raised by companies in India
avg_funds_raised = india_df['Funds_Raised'].mean()

# Total funds raised by companies in India
total_funds_raised = india_df['Funds_Raised'].sum()

num_companies, total_layoffs, avg_funds_raised, total_funds_raised


(143, 41980.0, 1071.4362573099415, 183215.6)

In [25]:
# Top industries in India affected by layoffs
top_industries = india_df.groupby('Industry')['Laid_Off_Count'].sum().sort_values(ascending=False).head(10)

top_industries


Industry
Education         12029.0
Food               5465.0
Transportation     4770.0
Retail             4146.0
Finance            3807.0
Consumer           3028.0
Travel             2500.0
Healthcare         1647.0
Fitness            1100.0
Logistics           680.0
Name: Laid_Off_Count, dtype: float64

In [26]:
# Distribution of layoffs across different stages of companies in India
company_stages_distribution = india_df.groupby('Stage')['Laid_Off_Count'].sum().sort_values(ascending=False)

company_stages_distribution


Stage
Unknown           6816.0
Private Equity    5040.0
Series J          3570.0
Series D          3301.0
Acquired          3260.0
Series B          2393.0
Series I          2250.0
Series E          2159.0
Series G          2100.0
Series H          2100.0
Series F          2001.0
Series C          1851.0
Series A          1750.0
Post-IPO          1650.0
Subsidiary        1300.0
Seed               439.0
Name: Laid_Off_Count, dtype: float64

In [27]:
# Companies in India with the highest number of layoffs
top_companies_layoffs = india_df[['Company', 'Laid_Off_Count']].sort_values(by='Laid_Off_Count', ascending=False).head(10)

top_companies_layoffs


Company  Laid_Off_Count
1268            Byju's          2500.0
2103       WhiteHat Jr          1800.0
2094         Bytedance          1800.0
530             Byju's          1500.0
2209       PaisaBazaar          1500.0
2256               Ola          1400.0
2268            Swiggy          1100.0
98    Reliance JioMart          1000.0
2020         Unacademy          1000.0
45              Byju's          1000.0

**Companies with Highest Layoffs:**

* Byju's: 2,500 layoffs
* WhiteHat Jr: 1,800 layoffs
* Bytedance: 1,800 layoffs
* Byju's: 1,500 layoffs (Another instance, likely from a different time period)
* PaisaBazaar: 1,500 layoffs
* Ola: 1,400 layoffs
* Swiggy: 1,100 layoffs
* Reliance JioMart: 1,000 layoffs
* Unacademy: 1,000 layoffs
* Byju's: 1,000 layoffs (Yet another instance, indicating multiple instances of layoffs)
 
> It's evident that some of the prominent companies in the edtech sector, such as Byju's, WhiteHat Jr, and Unacademy, have faced significant layoffs. This is in line with our earlier observation that the Education sector had the highest number of layoffs in India.

In [28]:
# Trend of layoffs in India over time
india_layoffs_trend = india_df.groupby('Date')['Laid_Off_Count'].sum().reset_index()

# Plotting the trend
fig_trend = px.line(india_layoffs_trend, 
                    x='Date', 
                    y='Laid_Off_Count', 
                    title='Trend of Layoffs in India Over Time',
                    labels={'Date': 'Date', 'Laid_Off_Count': 'Number of Layoffs'})

fig_trend.show()


**In summary, based on the data:**

* India saw significant layoffs across various industries, with the education sector being the most affected.
* Prominent companies in the edtech sector had a considerable number of layoffs.
* The trend over time would give insights into any patterns or specific periods with increased layoffs.

In [29]:
# Funds raised by the top companies with the highest layoffs
top_companies_funds = india_df[india_df['Company'].isin(top_companies_layoffs['Company'])][['Company', 'Funds_Raised']]
top_companies_funds = top_companies_funds.drop_duplicates().sort_values(by='Funds_Raised', ascending=False)

top_companies_funds


Company  Funds_Raised
2094         Bytedance        7400.0
45              Byju's        5500.0
737                Ola        5000.0
2256               Ola        3800.0
653             Swiggy        3600.0
2167            Swiggy        1600.0
274          Unacademy         838.0
2209       PaisaBazaar         496.0
1750       WhiteHat Jr          11.0
98    Reliance JioMart           NaN

**Funds Raised by Top Companies:**

* Bytedance: $7.4 billion
* Byju's: $5.5 billion
* Ola: $5 billion (This includes funds raised during different funding rounds)
* Swiggy: $5.2 billion (This includes funds raised during different funding rounds)
* Unacademy: $838 million
* PaisaBazaar: $496 million
* WhiteHat Jr: $11 million
* Reliance JioMart: Data not available

> It's evident that many of these companies had substantial funds raised, suggesting that factors other than immediate financial constraints might have influenced the layoffs.

In [30]:
# Distribution of layoffs across different company stages in India
stage_avg_layoffs = india_df.groupby('Stage')['Laid_Off_Count'].mean().sort_values(ascending=False)
stage_median_layoffs = india_df.groupby('Stage')['Laid_Off_Count'].median().sort_values(ascending=False)

stage_layoffs_analysis = pd.DataFrame({
    'Average Layoffs': stage_avg_layoffs,
    'Median Layoffs': stage_median_layoffs
}).reset_index()

stage_layoffs_analysis


Stage  Average Layoffs  Median Layoffs
0         Acquired       326.000000           180.0
1         Post-IPO       412.500000           350.0
2   Private Equity      1260.000000          1250.0
3             Seed       109.750000           120.0
4         Series A       102.941176            70.0
5         Series B       132.944444            90.0
6         Series C       142.384615           120.0
7         Series D       194.176471           142.0
8         Series E       166.076923           100.0
9         Series F       333.500000           275.5
10        Series G      1050.000000          1050.0
11        Series H       420.000000           350.0
12        Series I       750.000000           800.0
13        Series J       510.000000           200.0
14      Subsidiary       650.000000           650.0
15         Unknown       219.870968           120.0

**Interestingly, companies in the "Private Equity" and "Series G" stages had higher average layoffs compared to others.**

In [31]:
# Identifying the months with the highest layoffs in India
india_df['Month'] = pd.to_datetime(india_df['Date']).dt.to_period('M')
monthly_layoffs = india_df.groupby('Month')['Laid_Off_Count'].sum().sort_values(ascending=False).head(10)

monthly_layoffs


/tmp/ipykernel_20/1803304155.py:2: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



Month
2020-05    6361.0
2021-01    3600.0
2022-10    2743.0
2023-01    2327.0
2023-02    2280.0
2022-06    2261.0
2023-04    2222.0
2022-05    2109.0
2020-04    2051.0
2020-06    2020.0
Freq: M, Name: Laid_Off_Count, dtype: float64

**Months with Highest Layoffs:**

* May 2020: 6,361 layoffs
* January 2021: 3,600 layoffs
* October 2022: 2,743 layoffs
* January 2023: 2,327 layoffs
* February 2023: 2,280 layoffs
* June 2022: 2,261 layoffs
* April 2023: 2,222 layoffs
* May 2022: 2,109 layoffs
* April 2020: 2,051 layoffs
* June 2020: 2,020 layoffs

In [32]:
# Companies in India with the highest number of layoffs
top_companies_by_layoffs = india_df.groupby('Company')['Laid_Off_Count'].sum().sort_values(ascending=False).head(10)

top_companies_by_layoffs


Company
Byju's              5000.0
Swiggy              2880.0
Ola                 2800.0
WhiteHat Jr         2100.0
Bytedance           1800.0
Unacademy           1500.0
PaisaBazaar         1500.0
OYO                 1200.0
Vedantu             1109.0
Reliance JioMart    1000.0
Name: Laid_Off_Count, dtype: float64

**Top Companies by Layoffs:**

* Byju's: 5,000 layoffs
* Swiggy: 2,880 layoffs
* Ola: 2,800 layoffs
* WhiteHat Jr: 2,100 layoffs
* Bytedance: 1,800 layoffs
* Unacademy: 1,500 layoffs
* PaisaBazaar: 1,500 layoffs
* OYO: 1,200 layoffs
* Vedantu: 1,109 layoffs
* Reliance JioMart: 1,000 layoffs

In [33]:
# Companies in India with high funds raised and significant layoffs
top_funded_companies_with_layoffs = india_df.sort_values(by=['Funds_Raised', 'Laid_Off_Count'], ascending=[False, False]).head(10)

top_funded_companies_with_layoffs[['Company', 'Funds_Raised', 'Laid_Off_Count', 'Industry', 'Stage', 'Date']]


Company  Funds_Raised  Laid_Off_Count        Industry  \
2247          Uber       24700.0           600.0  Transportation   
2269        WeWork       19500.0           100.0     Real Estate   
373       Flipkart       12900.0             NaN          Retail   
479   TikTok India        9400.0            40.0        Consumer   
2027           Sea        8600.0           350.0          Retail   
2094     Bytedance        7400.0          1800.0        Consumer   
1268        Byju's        5500.0          2500.0       Education   
530         Byju's        5500.0          1500.0       Education   
45          Byju's        5500.0          1000.0       Education   
1564           Ola        5000.0          1000.0  Transportation   

               Stage        Date  
2247        Post-IPO  2020-05-26  
2269        Series H  2020-05-18  
373         Acquired  2023-03-02  
479         Acquired  2023-02-10  
2027        Post-IPO  2022-03-30  
2094         Unknown  2021-01-27  
1268  Private Equity  2022-10-12  
530   Private Equity  2023-02-02  
45    Private Equity  2023-06-07  
1564        Series J  2022-07-29

**From this, we can observe:**

* Uber, despite being a post-IPO company and having raised a significant amount of funds, had 600 layoffs in May 2020.
* Bytedance, the parent company of TikTok, faced 1,800 layoffs in January 2021.
* Byju's, an education company that raised a considerable amount, reported multiple instances of layoffs, with the most significant one being 2,500 layoffs in October 2022.

> The data suggests that while financial backing is crucial, other factors, possibly operational or industry-specific challenges, could have influenced the decision to lay off employees.

In [34]:
# Filtering the dataset for entries from the year 2023
india_2023_df = india_df[india_df['Date'].str.contains('2023')]

# Industry-wise analysis of layoffs and funds for 2023

# Average and total layoffs by industry for 2023
industry_2023_layoffs = india_2023_df.groupby('Industry')['Laid_Off_Count'].agg(['mean', 'sum']).rename(columns={'mean': 'Average Layoffs', 'sum': 'Total Layoffs'})

# Average and total funds raised by industry for 2023
industry_2023_funds = india_2023_df.groupby('Industry')['Funds_Raised'].agg(['mean', 'sum']).rename(columns={'mean': 'Average Funds ($M)', 'sum': 'Total Funds ($M)'})

# Merging the two dataframes
industry_2023_analysis = industry_2023_layoffs.join(industry_2023_funds)

industry_2023_analysis.sort_values(by='Total Layoffs', ascending=False).head(10)


Average Layoffs  Total Layoffs  Average Funds ($M)  \
Industry                                                             
Education            401.111111         3610.0            1294.800   
Food                 319.166667         1915.0             666.250   
Retail               141.333333         1272.0            1709.000   
Healthcare           148.500000          891.0             293.875   
Consumer             179.500000          718.0            2800.000   
Finance               85.857143          601.0              69.000   
Other                250.000000          500.0             689.500   
Support              128.500000          257.0              57.500   
Transportation       120.000000          240.0            1340.000   
Crypto                90.000000          180.0             347.500   

                Total Funds ($M)  
Industry                          
Education                12948.0  
Food                      5330.0  
Retail                   15381.0  
Healthcare                2351.0  
Consumer                 11200.0  
Finance                    552.0  
Other                     1379.0  
Support                    115.0  
Transportation            5360.0  
Crypto                     695.0

**In 2023:**

* The Education industry led in terms of layoffs, with a total of 3,610 layoffs, even though it also had a substantial funding amount of approximately $12.9 billion.

* The Food and Retail industries followed, with 1,915 and 1,272 layoffs, respectively.

* Interestingly, the Consumer industry, despite having a significant funding amount of $11.2 billion, faced 718 layoffs.

In [35]:
import plotly.graph_objects as go

# Visualization for Industry-wise Layoffs in 2023

# Sorting the data
industry_2023_layoffs_sorted = industry_2023_analysis.sort_values(by='Total Layoffs', ascending=False)

# Bar chart for Industry-wise Layoffs
fig1 = go.Figure(data=[
    go.Bar(name='Total Layoffs', x=industry_2023_layoffs_sorted.index, y=industry_2023_layoffs_sorted['Total Layoffs'])
])

# Update layout for better visualization
fig1.update_layout(
    title='Industry-wise Layoffs in 2023',
    xaxis_title='Industry',
    yaxis_title='Number of Layoffs',
    xaxis={'categoryorder':'total descending'},
    barmode='group'
)

fig1.show()


In [36]:
# Sorting the data for funds
industry_funds_sorted = industry_2023_analysis.sort_values(by='Total Funds ($M)', ascending=False)

# Bar chart for Industry-wise Funds Raised
fig2 = go.Figure(data=[
    go.Bar(name='Total Funds ($M)', x=industry_funds_sorted.index, y=industry_funds_sorted['Total Funds ($M)'])
])

# Update layout for better visualization
fig2.update_layout(
    title='Industry-wise Funds Raised in 2023',
    xaxis_title='Industry',
    yaxis_title='Funds Raised ($M)',
    xaxis={'categoryorder':'total descending'},
    barmode='group'
)
fig2.show()


In [37]:
# Convert the 'Month' column to string to avoid serialization issues
india_2023_df['Month'] = india_2023_df['Month'].astype(str)

# Grouping by month for layoffs trend
monthly_layoffs_2023 = india_2023_df.groupby('Month')['Laid_Off_Count'].sum()

# Line chart for Monthly Layoffs Trend using Scatter
fig3 = go.Figure(data=[
    go.Scatter(x=monthly_layoffs_2023.index, y=monthly_layoffs_2023.values, mode='lines+markers')
])

# Update layout for better visualization
fig3.update_layout(
    title='Monthly Layoffs Trend in 2023',
    xaxis_title='Month',
    yaxis_title='Number of Layoffs'
)
fig3.show()


/tmp/ipykernel_20/4121689979.py:2: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



In [38]:
# Grouping by month and company for layoffs trend in 2023
monthly_company_layoffs_2023 = india_2023_df.groupby(['Month', 'Company'])['Laid_Off_Count'].sum().unstack()

# Line chart for Monthly Layoffs Trend by Company using Scatter
fig4 = go.Figure()

# Adding a line for each company
for company in monthly_company_layoffs_2023.columns:
    fig4.add_trace(go.Scatter(x=monthly_company_layoffs_2023.index, y=monthly_company_layoffs_2023[company], mode='lines+markers', name=company))

# Update layout for better visualization
fig4.update_layout(
    title='Monthly Layoffs Trend by Company in 2023',
    xaxis_title='Month',
    yaxis_title='Number of Layoffs',
    legend_title_text='Company'
)

fig4.show()


In [39]:
import plotly.express as px

# Grouping by company for total layoffs in 2023
company_layoffs_2023 = india_2023_df.groupby('Company')['Laid_Off_Count'].sum().sort_values(ascending=False)

# Bar chart for Company-wise Total Layoffs in 2023
fig5 = px.bar(company_layoffs_2023, x=company_layoffs_2023.index, y=company_layoffs_2023.values, title="Company-wise Total Layoffs in 2023")

# Updating layout for better visualization
fig5.update_layout(
    xaxis_title='Company',
    yaxis_title='Number of Layoffs',
    xaxis={'categoryorder':'total descending'}
)

fig5.show()
